Fix ambient colour being black not rendering point lights.

isidornygren opened the pull request:
## Summary

Uses the ambient light colour as the base for point light rendering instead of multiplying it later with the point light colour. This renders without issues if the ambient colour is black.

Fixes https://github.com/jgayfer/bevy_light_2d/issues/22

jgayfer reviewed (commented):
Hi, thanks for the PR!

While I believe the proposed change is the correct way to approach this problem (from an objective standpoint), it does make lights much brighter than they were prior.

| Before | After |
| - | - |
| <img width="441" alt="Screenshot 2024-08-17 at 2 29 13 PM" src="https://github.com/user-attachments/assets/18f3f15a-9ad7-4463-a31d-eedb8b7cf74c"> | <img width="455" alt="Screenshot 2024-08-17 at 2 28 50 PM" src="https://github.com/user-attachments/assets/18ce297d-1db1-446c-974a-b316cd0561da"> |

This can be fixed in the `dungeon` example by knocking the light intensity from `25.0` to `2.0`, which _is_ a bit more of a sane value, from a "numbers" point of view.

I like the change, but am mostly unsure what to do from a release point of view, as this would be breaking. I suppose a `0.4` release is the answer for that.

jgayfer reviewed (commented):
We should probably:

- Update the `intensity` value on point lights in our examples to keep the same look (I believe this is just the [`dungeon`](https://github.com/jgayfer/bevy_light_2d/blob/main/examples/dungeon.rs) example)
- Add an entry to [`CHANGELOG.md`](https://github.com/jgayfer/bevy_light_2d/blob/main/CHANGELOG.md) under the `Unreleased` heading documenting this change, with a reference to this PR number (`#24`)

isidornygren commented:
Howdy, and thanks for the quick response!

> We should probably:
>
> * Update the `intensity` value on point lights in our examples to keep the same look (I believe this is just the [`dungeon`](https://github.com/jgayfer/bevy_light_2d/blob/main/examples/dungeon.rs) example)
> 
> * Add an entry to [`CHANGELOG.md`](https://github.com/jgayfer/bevy_light_2d/blob/main/CHANGELOG.md) under the `Unreleased` heading documenting this change, with a reference to this PR number (`#24`)
> 
> 
> Then we can move ahead with the PR. Thank you!

Absolutely. I've updated the branch with updated intensity values for the `dungeon` and the `multiple` examples as they seemed like the only ones affected. I opted for an intensity value of 1.0 in the `multiple` example due to it being a nicer value than `0.5` (which is the value that more closely renders similarly to prior the change), I think it still looks acceptable though:

| Example | Before | After |
| --- | --- | --- |
| Basic | <img width="1392" alt="Screenshot 2024-08-18 at 10 04 07" src="https://github.com/user-attachments/assets/384b51b5-6b4e-402e-9266-d70fe7bd2b8f"> | <img width="1392" alt="Screenshot 2024-08-18 at 10 11 06" src="https://github.com/user-attachments/assets/c4801135-450b-4aef-8bea-e76855a57f48"> |
| Dungeon | <img width="1392" alt="Screenshot 2024-08-18 at 10 03 56" src="https://github.com/user-attachments/assets/a7dbf249-a37a-45bc-8df7-d5448b62e891"> | <img width="1392" alt="Screenshot 2024-08-18 at 10 09 45" src="https://github.com/user-attachments/assets/ea7d03b0-8cf9-4210-a3ff-a061e8b9f262"> |
| Multiple | <img width="1392" alt="Screenshot 2024-08-18 at 10 14 16" src="https://github.com/user-attachments/assets/f856939f-eee2-4f94-b41c-1673c970c820"> | <img width="1392" alt="Screenshot 2024-08-18 at 10 17 01" src="https://github.com/user-attachments/assets/2e99c35b-78dc-4f19-8b9d-1c1eb925aeff"> |
| Occlusion | <img width="1392" alt="Screenshot 2024-08-18 at 10 04 45" src="https://github.com/user-attachments/assets/cfb150e6-a85c-493c-a1eb-2d545c50d79e"> | <img width="1392" alt="Screenshot 2024-08-18 at 10 13 30" src="https://github.com/user-attachments/assets/474f151f-0588-48fd-9b5f-3101164e353f"> |

I've updated the changelog with a link to this PR, but didn't mention the breaking change. I'm not sure if that should be included in the changelog or communicated somewhere else.

jgayfer commented:
Makes sense, thanks!

On the breaking change, maybe we should add some other entries to the change log?

```md
## Changed

- Point lights colours are now _added_ to ambient light, instead of multiplied by it. 

## Migration guide

- As point light colour is now added to ambient light,instead of multiplied by it, ambient lights with brightness less than zero, point lights will need to have their `intensity` value toned down to maintain the same behaviour.
```

isidornygren commented:
> Makes sense, thanks!
> 
> On the breaking change, maybe we should add some other entries to the changelog?
> 
> ```
> ## Changed
> 
> - Point lights colours are now added to ambient light, instead of multiplied by it (#24). 
> 
> ## Migration guide
> 
> - Point light intensity needs to be adjusted to account for changes to ambient light. Generally this means point light intensity values need to be lowered. See the relevant changes to the `dungeon` example.
> ```

Perfection, I added your suggestion as-is.

jgayfer reviewed (approved):
Looks good. Thank you!
